File: Python/number.py
from bangtal import *
from config import *

class Number(Object):
    def __init__(self, scene, x, y, shown = True):
        super().__init__(Config.IMAGE_PATH + "0.png")

        self.locate(scene, x, y)
        if shown: self.show()

    def setNumber(self, n):
        if n < 0 or n > 9: self.hide()
        else:
            self.setImage(Config.IMAGE_PATH + str(n) + ".png")
            self.show()

File: C++/Number.cpp
#include "Number.h"
using namespace bangtal;

#include "config.h"

#include <string>
using namespace std;

NumberPtr Number::create(ScenePtr scene, int x, int y, bool shown)
{
	auto number = NumberPtr(new Number(scene, x, y, shown));
	Object::add(number);

	return number;
}

Number::Number(ScenePtr scene, int x, int y, bool shown)
	: Object("", scene, x, y, shown)
{
}

void Number::setNumber(int n)
{
	if (n >= 0 && n < 10) {
#ifdef WIN32
		string image = IMAGE_PATH + to_string(n) + ".png";
#else
		string image = to_string(n) + ".png";
#endif
		setImage(image);
		show();
	}
	else {
		setImage("");
		hide();
	}
}
